internal/buildkitd/buildkitd.go

```diff
@@ -67,259 +63,155 @@ func Client(ctx context.Context) (*bkclient.Client, error) {
-		"run",
-		"-d",
-		"--restart", "always",
-		"-v", volumeName+":/var/lib/buildkit",
-		"--name", containerName,
-		"--privileged",
-		image+":"+version,
-		"--debug",
-	)
-	output, err := cmd.CombinedOutput()
-	if err != nil {
-		// If the daemon failed to start because it's already running,
-		// chances are another dagger instance started it. We can just ignore
-		// the error.
-		if !strings.Contains(string(output), "Error response from daemon: Conflict.") {
-			return err
-		}
-	}
-	return waitBuildkit(ctx)
-}
-
-// waitBuildkit waits for the buildkit daemon to be responsive.
-func waitBuildkit(ctx context.Context) error {
-	c, err := bkclient.New(ctx, "docker-container://"+containerName)
-	if err != nil {
-		return err
+// TODO: move to own package (duplicate with version.go)
+// revision returns the VCS revision being used to build or empty string
+// if none.
+func revision() (string, error) {
+	bi, ok := debug.ReadBuildInfo()
+	if !ok {
+		return "", errors.New("unable to read build info")
 	}
-
-	// FIXME Does output "failed to wait: signal: broken pipe"
-	defer c.Close()
-
-	// Try to connect every 100ms up to 100 times (10 seconds total)
-	const (
-		retryPeriod   = 100 * time.Millisecond
-		retryAttempts = 100
-	)
-
-	for retry := 0; retry < retryAttempts; retry++ {
-		_, err = c.ListWorkers(ctx)
-		if err == nil {
-			return nil
+	for _, s := range bi.Settings {
+		if s.Key == "vcs.revision" {
+			return s.Value[:9], nil
 		}
-		time.Sleep(retryPeriod)
-	}
-	return errors.New("buildkit failed to respond")
-}
-
-func removeBuildkit(ctx context.Context) error {
-	cmd := exec.CommandContext(ctx,
-		"docker",
-		"rm",
-		"-fv",
-		containerName,
-	)
-	_, err := cmd.CombinedOutput()
-	if err != nil {
-		return err
 	}
-
-	return nil
+	return "", nil
 }
 
-func getBuildkitInformation(ctx context.Context) (*buildkitInformation, error) {
-	formatString := "{{.Config.Image}};{{.State.Running}};{{if index .NetworkSettings.Networks \"host\"}}{{\"true\"}}{{else}}{{\"false\"}}{{end}}"
-	cmd := exec.CommandContext(ctx,
-		"docker",
-		"inspect",
-		"--format",
-		formatString,
-		containerName,
-	)
-	output, err := cmd.CombinedOutput()
-	if err != nil {
-		return nil, err
-	}
-
-	s := strings.Split(string(output), ";")
-
-	// Retrieve the tag
-	ref, err := reference.ParseNormalizedNamed(strings.TrimSpace(s[0]))
-	if err != nil {
-		return nil, err
-	}
-	tag, ok := ref.(reference.Tagged)
-	if !ok {
-		return nil, fmt.Errorf("failed to parse image: %s", output)
-	}
-
-	// Retrieve the state
-	isActive, err := strconv.ParseBool(strings.TrimSpace(s[1]))
-	if err != nil {
-		return nil, err
-	}
-
-	return &buildkitInformation{
-		Version:  tag.Tag(),
-		IsActive: isActive,
-	}, nil
+type Provisioner interface {
+	RemoveDaggerd(ctx context.Context) error
+	InstallDaggerd(ctx context.Context, version string) error
+	StartDaggerd(ctx context.Context) error
+	GetBuildkitInformation(ctx context.Context) (*buildkitInformation, error)
+	GetHost() string
```

Comment:
nit: can we combine these two methods into something like `DaggerdState(ctx context.Context)`? I got confused by what `GetHost` was doing and we can remove direct references to Buildkit now that we are wrapping.

Not a blocker if hard for whatever reason.